src/azure-cli/azure/cli/command_modules/resource/_help.py

```diff
@@ -1271,131 +1272,167 @@
 examples:
-  - name: Delete a resource group lock
+  - name: Delete a resource group lock using the resource group name. The resource group name and lock name are both required if the `--ids` parameter is not used.
     text: >
-        az group lock delete --name lockName -g MyResourceGroup
+        az group lock delete --name MyLockName --resource-group MyResourceGroup
+  - name: Delete a one or more resource group locks using the lock Id. For a list of available lock Ids, run `az group lock list --resource-group MyResourceGroup -o json`.
+    text: >
+        az group lock delete --ids /subscriptions/MySubscriptionID/resourceGroups/MyResourceGroup/providers/Microsoft.Authorization/locks/MyLockName
 """
 
 helps['group lock list'] = """
 type: command
-short-summary: List lock information in the resource-group.
+short-summary: List lock information in a resource-group.
+long-summary: To list all locks for a subscription, regardless of resource group, use `az lock list`.
 examples:
-  - name: List out all locks on the resource group level
+  - name: List all locks for a resource group.
     text: >
-        az group lock list -g MyResourceGroup
+        az group lock list --resource-group MyResourceGroup
+  - name: List all locks meeting a filter criteria. For more information on using the OData `$filter` query parameter, see [Use the $filter query parameter](https://learn.microsoft.com/graph/filter-query-parameter)
+        az group lock list --filter-string "name eq 'MyLockName'"
+  - name: List all locks meeting a `--query` criteria.
+        az group lock list --resource-group MyResourceGroup --query "[?level=='ReadOnly']"
 """
 
 helps['group lock show'] = """
 type: command
-short-summary: Show the details of a resource group lock
+short-summary: Show the details of a resource group lock.
 examples:
-  - name: Show a resource group level lock
+  - name: Show the details of a resource group lock. The resource group name and lock name are both required if the `--ids` parameter is not used.
+    text: >
+        az group lock show --name MyLockName --resource-group MyResourceGroup
+  - name: >
+      Show the details of a many resource group locks using the Id.
+      The `--resource-group` and `--name` parameters are not needed in this example as these values exist in the ID.
+      For a list of available lock Ids, run `az group lock list --resource-group MyResourceGroup -o json`.
     text: >
-        az group lock show -n lockname -g MyResourceGroup
+        az group lock show --ids /subscriptions/MySubscriptionID/resourceGroups/MyResourceGroup/providers/Microsoft.Authorization/locks/MyLockName1 \\
+            /subscriptions/MySubscriptionID/resourceGroups/MyResourceGroup/providers/Microsoft.Authorization/locks/MyLockName2 
 """
 
 helps['group lock update'] = """
 type: command
-short-summary: Update a resource group lock.
+short-summary: Update a resource group lock with new notes, or change the lock type.
```

Comment:
I'm not sure whether other properties will be expanded in the future. Do we need to describe the `short-summary` so specifically?